src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -1780,7 +1780,16 @@ theorem extractLsb'_mul {w len} {x y : BitVec w} (hlen : len ≤ w) :
 
 /-- Adding bitvectors that are zero in complementary positions equals concatenation. -/
 theorem append_add_append_eq_append {v w : Nat} {x : BitVec v} {y : BitVec w} :
-    (x ++ 0#w) + (0#v ++ y) = x ++ y := by
+    (HAppend.hAppend (α := BitVec (no_index _)) (β := BitVec (no_index _)) (γ := BitVec (no_index _)) x 0#w) + 
+    (HAppend.hAppend (α := BitVec (no_index _)) (β := BitVec (no_index _)) (γ := BitVec (no_index _)) 0#v y)
```

Comment:
IIUC, there's no need to `no_index` the input types.
```suggestion
    (HAppend.hAppend (γ := BitVec (no_index _)) x 0#w) + 
    (HAppend.hAppend (γ := BitVec (no_index _)) 0#v y)
```

It might be good to add a comment explaining why `no_index` is needed, and primarily, why there is no need to `no_index` the occurrences of these types in `HAdd.hAdd`

Reply:
Done!